src/renderers/DomRenderer.js

```diff
@@ -248,6 +248,14 @@ function insertStyle(style: HTMLElement, options: PriorityOptions) {
   getHead().insertBefore(style, prevNode)
 }
 
+/**
+ * Read jss nonce setting from the page if the user has set it.
+ */
+function getNonceFromPage() : ? string {
```

Comment:
Right now you are running this query when stylesheet is created each time, its a bit of unnecessarily, we can cache it similar to getHead. Also as this is the second place we need memoization, its worth adding a simple local memoze helper:

```js
const memoize = (fn) => {
  let value
  return () => {
    if (!value) value = fn()
    return value
  }
}
```

```js
const getNonce = memoize(() => {
   ....return the nonce
})

const getBody = memoize(() => {
   ....return the body
})
```

Reply:
Can do.

Reply:
I've cached the value as you suggested.

However, Flow was giving me a bunch of trouble with that `memoize` function. It probably requires generics or something and I'm really not familiar with Flow. Perhaps this can be done as an future enhancement  by someone who is more familiar with it.

Reply:
just don't add typings to memoize

Reply:
we should be adding typings anyways to public api only

Reply:
Even without adding typings, the type inference causes problems.

Reply:
does using any solve it?

Reply:
I'm going to say it's better to repeat some trivial code twice rather than lose the type checking. If this was being done 3+ times then maybe

Reply:
I can check it out if you commit with --no-verify

Reply:
there you go